rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+                event.location_label == sender_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Submit {collateral_amount} {A_ETH.symbol_or_name()} to Lido for receiving {A_STETH.symbol_or_name()} in exchange'  # noqa: E501
+                event.counterparty = CPT_LIDO_ETH
+                #  preparing next action to be processed when erc20 transfer will be decoded by rotki  # noqa: E501
+                #  needed because submit levent is emitted prior of erc20 transfer, so it is not decoded yet  # noqa: E501
+                # TODO: to be confirmed with ROTKI team if it is not possible to have the erc20 event available before this decoder is called  # noqa: E501
+                paired_event = event
+                action_from_event_type = HistoryEventType.RECEIVE
+                action_to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                action_to_notes = f'Receive {{amount}} {A_STETH.symbol_or_name()} in exchange of the deposited {A_ETH.symbol_or_name()}'  # {amount} to be replaced in post decoding  # noqa: E501
+                break
+
+        action_items = []  # also create an action item for the reception of the stETH tokens
+        if paired_event is not None and action_from_event_type is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=action_from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=A_STETH,
+                amount=steth_minted_tokens,
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_LIDO_ETH,
+                paired_event_data=(paired_event, True),
+                extra_data={'staked_eth': str(collateral_amount)},
+            ))
+        else:  # did not break/find anything
+            log.error(
+                'At lido steth submit decoding of tx '
+                'not found the decoded event of the ETH transfer',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO_ETH)
+
+    def _map_erc20_lido_token_transfer(
+            self,
+            decoded_events: list['EvmEvent'],
+            transaction: 'EvmTransaction',  # pylint: disable=unused-argument
+            all_logs: list['EvmTxReceiptLog'],  # pylint: disable=unused-argument
+    ) -> list['EvmEvent']:
+        """Post decoding function to map the ERC20 Lido tokens reception"""
+        """TODO: To be checked with Rotki team, I am adding post decoding because I see that ERC20
+        transfers are not available in decoded_events at the time of decode by address rules
+        from what I have seen in the code. I got this solution looking at compound rules"""
```

Comment:
Never add two docstrings.

```suggestion
        """TODO: To be checked with Rotki team, I am adding post decoding because I see that ERC20
        transfers are not available in decoded_events at the time of decode by address rules
        from what I have seen in the code. I got this solution looking at compound rules"""
```

I don't understand what this is supposed to do. As I said post decoding rules should be avoided like the plague as they are really slow and are adding extra steps in our event processing flow.

Reply:
As I reported in the comment, I was forced to use this solution because "submitted" log event happens before the erc20 transfer, and the decoded event erc20 transfer is not yet processed by rotki when the decoder for "submitted" is run. Also, basically I used exactly the same mechanism used in compound_v3 rules that was used for solving the same problem I think: https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/evm/decoding/compound/v3/decoder.py#L388